test_data/asm/expand_fixed.asm

```diff
@@ -0,0 +1,101 @@
+use std::prover::degree;
+use std::check::assert;
+
+// A ArrayTerm in an `ArrayExpression`
+enum ArrayTerm {
+    Repeat(int[]),
+    Once(int[])
+}
+
+// returns the total size of the repeated array in this `ArrayExpression`
+let solve: ArrayTerm[] -> int = |terms| {
+    let (_, total_size_of_non_repeated) = std::array::fold(terms, (false, 0), |(found_repeated, len), term| {
+        match term {
+            // we can have `[]*` as many times as we want
+            ArrayTerm::Repeat([]) => (found_repeated, len),
+            ArrayTerm::Repeat(a) => {
+                if found_repeated {
+                    std::check::panic("unsolvable because two terms are repeated")
+                } else {
+                    (true, len)
+                }
+            },
+            ArrayTerm::Once(a) => (found_repeated, len + std::array::len(a))
+        }
+    });
+    //std::check::assert(total_size_of_non_repeated <= degree(), || "non repeated array terms do not fit in degree");
```

Comment:
@chriseth if I uncomment this line, I get `Tried to add constraints outside of statement context`

Reply:
I think this is a bug in the evaluator: If a statement evaluates to `()`, it should not add a constraint.

Reply:
Here is the fix: https://github.com/powdr-labs/powdr/pull/1568